### ISSUE TITLE
[Help needed] Artistic Network Architecture

### ISSUE THREAD

**Issue by GlebBrykin** [jantic/DeOldify#319]
Hi, Jason!
Your work is truly remarkable! I really liked the results of your artistic DeOldify and would like to port it to C# to help Windows users. I would like to ask you to show the neural network architecture so that I can work with the weights correctly.
Sincerely,
Gleb Brykin

**Comment by jantic** [comment]
@GlebBrykin  Your best bet to port to C# would probably be to export to ONNX, and then use something like Microsoft's  [onnxruntime](https://github.com/microsoft/onnxruntime) to execute it in C#.  To start, something like this to export it:
`
import os
from deoldify.generators import gen_inference_deep
import torch.nn as nn
from pathlib import Path
from fastai.vision.data import normalize_funcs, imagenet_stats

norm, denorm = normalize_funcs(*imagenet_stats)
sz =  (560,560) 

class ImageScaleInput(nn.Module):
    def __init__(self): 
        super().__init__()
        self.norm = norm

    def forward(self, x): 
        out = (x.div(255.0)).type(torch.float32)
        out,_ = self.norm((out,out), do_x=True)
        #out = out.unsqueeze(0)
        return out

class ImageScaleOutput(nn.Module):
    def __init__(self): 
        super().__init__()
        self.denorm = denorm

    def forward(self, x):
        out = self.denorm(x, do_x=True)
        out = out.float().clamp(min=0,max=1) 
        out = self.denorm(out, do_x=False)
        out = (out.mul(255.0)).type(torch.float32)
        return out
onnx_path = './models/ColorizeArtistic_gen.onnx'
raw_model = gen_inference_deep(root_folder=Path('./'), weights_name='ColorizeArtistic_gen').model

dummy_input = torch.randn(1, 3, sz[0], sz[1])
final_pytorch_model = nn.Sequential(ImageScaleInput(), raw_model , ImageScaleOutput())
torch.onnx.export(final_pytorch_model, dummy_input, onnx_path, input_names=['input'], output_names=['output'], opset_version=9)
`
I didn't test this particular snippet but this should be a good start- I've done this before.  Keep in mind: There's still work to be done on the C# side in terms of the image processing that's done in the Python code.  I won't be providing that but there's quite a bit involved in that including mapping just the hue/saturation of the model output image to the luminance of the original image.  This allows for processing of high definition images without losing image resolution.  It won't be totally easy in summary, but it can be done.

**Comment by navyasingh609** [comment]
If you are looking for an <a href="https://www.nightangels.in/escorts/jaipur">Escort Service in Jaipur</a> then you have come to the right place, VIP Escorts is one of the trusted escort service providers in Jaipur. We specialize in providing high-profile escort services to our elite clientele. You can book our escorts for an outcall or incall on any day of your choice and at any time of your choice.
<a href="https://www.nightangels.in/escorts/mumbai">Mumbai Escorts Services</a>
<a href="https://www.nightangels.in/escorts/bangalore">Bangalore Escort</a>

**Comment by navyasingh609** [comment]
Many people might still view <a href="https://www.ctgirls.in/">Mumbai Escort</a> in a red light. However, with so many firms offering their services, it opens up opportunities for people to indulge in their fantasies. With the right person by your side, you'll be able to enjoy everything from exotic dinners and movie screenings to a day at the spa or even on an ocean cruise.

**Comment by navyasingh609** [comment]
I am an <a href="https://www.dakshi.in/">Bangalore Escort</a> and I can offer you the most exciting escort services. You can contact me for booking without any hesitation because I will provide you unforgettable time with me. It doesn't matter if you are in Bangalore or outside this city, my service is available 24 hours a day 7 days a week for your convenience.
